core/01_core.js

```diff
@@ -221,59 +257,359 @@
         err.code = res.code;
       }
       // Strip unwrapOpResult() and errorBuilder() calls from stack trace
-      ErrorCaptureStackTrace(err, unwrapOpResult);
-      throw err;
+      ErrorCaptureStackTrace(err, hideFunction);
+      return PromiseReject(err);
     }
-    return res;
+    const promise = PromiseResolve(res);
+    promise[promiseIdSymbol] = id;
+    return promise;
+  }
+
+  /*
+Basic codegen.
+
+TODO(mmastrac): automate this (handlebars?)
+
+let s = "";
+const vars = "abcdefghijklm";
+for (let i = 0; i < 10; i++) {
+  let args = "";
+  for (let j = 0; j < i; j++) {
+    args += `${vars[j]},`;
+  }
+  s += `
+      case ${i}:
+        fn = function async_op_${i}(${args}) {
+          const id = nextPromiseId++;
+          try {
+            const maybeResult = originalOp(id, ${args});
+            if (maybeResult !== undefined) {
+              movePromise(id);
+              return unwrapOpResultNewPromise(id, maybeResult, async_op_${i});
+            }
+          } catch (err) {
+            movePromise(id);
+            ErrorCaptureStackTrace(err, async_op_${i});
+            return PromiseReject(err);
+          }
+          let promise = PromisePrototypeThen(setPromise(id), unwrapOpError(eventLoopTick));
+          promise = handleOpCallTracing(opName, id, promise);
+          promise[promiseIdSymbol] = id;
+          return promise;
+        };
+        break;
+  `;
+}
+  */
+
+  // This function is called once per async stub
+  function asyncStub(opName, args) {
+    setUpAsyncStub(opName);
+    return ops[opName].apply(undefined, args);
```

Comment:
Please use primordials

```suggestion
    return ReflectApply(ops[opName], undefined, args);
```